mysqld_exporter.go

```diff
@@ -918,8 +921,26 @@ func scrapeGlobalVariables(db *sql.DB, ch chan<- prometheus.Metric) error {
 }
 
 func scrapeSlaveStatus(db *sql.DB, ch chan<- prometheus.Metric) error {
-	slaveStatusRows, err := db.Query(slaveStatusQuery)
+	var (
+		slaveStatusRows *sql.Rows
+		err error
+	)
+	// Leverage lock-free SHOW SLAVE STATUS
+	if slaveStatusSuffix == "" {
```

Comment:
Scrapes can happen concurrently, so this block will have a race condition.

Perhaps find out the suffix upon start and not as part of the first scrape?